Author: Peter Lekensteyn <lekensteyn@gmail.com>
Origin: http://kmess.org/board/viewtopic.php?f=4&t=20549&start=15#p32892
Desciption: Follow Location redirects, MS seems to have moved some accounts to
 other servers. Forum topic reference:
 Connection problem to wlm - http://kmess.org/board/viewtopic.php?f=4&t=20549
--- a/src/network/soap/httpsoapconnection.cpp
+++ b/src/network/soap/httpsoapconnection.cpp
@@ -452,6 +452,7 @@ void HttpSoapConnection::slotRequestFini
   const QByteArray& replyContents( reply->readAll() );
   const int           statusCode = reply->attribute( QNetworkRequest::HttpStatusCodeAttribute   ).toInt();
   const QString&            error( reply->attribute( QNetworkRequest::HttpReasonPhraseAttribute ).toString() );
+  QUrl redirectUrl = reply->attribute( QNetworkRequest::RedirectionTargetAttribute ).toUrl();
 
 #ifdef KMESSDEBUG_HTTPSOAPCONNECTION_GENERAL
   bool requestSuccess = false;
@@ -486,13 +487,16 @@ void HttpSoapConnection::slotRequestFini
 #endif
 
     // Parse the message contents
-    if( currentResponse->isFaultMessage() )
+    if( !redirectUrl.isEmpty() || currentResponse->isFaultMessage() )
     {
-      // Verify if the server is redirecting us to another server
       if( currentResponse->getFaultCode() == "psf:Redirect" )
       {
+        redirectUrl = XmlFunctions::getNodeValue( currentResponse->getFault(), "redirectUrl" );
+      }
+      // Verify if the server is redirecting us to another server
+      if( !redirectUrl.isEmpty() )
+      {
         const QUrl& originalUrl( currentResponse->getEndPoint() );
-        const QUrl& redirectUrl( XmlFunctions::getNodeValue( currentResponse->getFault(), "redirectUrl" ) );
         const QString& originalHost( originalUrl.host() );
         const QString& redirectHost( redirectUrl.host() );
 
